lib/rules/indent.js

```diff
@@ -807,6 +889,84 @@ module.exports = {
             });
         }
 
+        /**
+         * Iterate through the dependent tokens until we reach the last token
+         * dependency or the last token dependency that is within the unknown
+         * node. If the last token dependency is not the first token of the line
+         * make the last token dependcy depend on the first token of the line.
+         * This ensures the offset will be set correctly since only the first
+         * token of each line is ignored and not all tokens on a line.
+         * @param {Token} dependency First
+         * @param {number} startLineOfNode First line number of unknown node
+         * @returns {void}
+         */
+        function ignoreUnknownNodeLineWithDependency(dependency, startLineOfNode) {
+            let currDependency = dependency,
+                lastDependency = dependency;
+
+            while (
+                (currDependency = offsets.getFirstDependency(currDependency)) &&
+                currDependency.loc.start.line > startLineOfNode
```

Comment:
(Continuing the discussion from https://github.com/eslint/eslint/pull/8504#discussion_r114679878 because that conversation is on an outdated diff)

Thanks for the detailed explanation, that was very useful.

One way to resolve this without iterating through the dependencies here would be to account for this case when the total desired indentation is being computed. I think it will work if you change [this line](https://github.com/eslint/eslint/blob/025e97a52ca31978c7fd731a7eefcff876393790/lib/rules/indent.js#L220) so that if a token with no offset dependency is not the first token of its line, the desired indentation of the first token of its line is used, rather than assuming that the desired indentation is 0.